src/mds/CInode.cc

```diff
@@ -3834,14 +3824,14 @@ void CInode::validate_disk_state(CInode::validated_data *results,
         shadow_in->fetch(get_internal_callback(INODE));
         return false;
       } else {
-        return fetch_dirfrag_rstats();
+	results->inode.passed = true;
```

Comment:
...oh, because you want to do a final check for passed_validation against each of these.

It'd be a little longer but safer to do the passed_validation check by only examining the ones which we checked. Eg
> (!results->raw_stats.checked || results->raw_stats.passed) &&
> (!results->backtrace.checked || results->backtrace.passed) &&
> (!results->inode.checked || results->inode.passed)

How about that instead of confusing the output semantics?